shared/types/guards.ts

```diff
@@ -83,3 +84,41 @@ export const isWalletFile = (wallet: any): wallet is WalletFile =>
   isWalletMeta(wallet.meta) &&
   (isWalletGCMEncrypted(wallet.crypto) ||
     isWalletLegacyEncrypted(wallet.crypto));
+
+export const validationWalletCipherTextDuplication = (
+  loadedWallets: WalletWithPath[],
+  cipherText: string
+): string => {
+  const duplicateWallet = loadedWallets.find(
+    (wallet) => wallet.wallet.crypto.cipherText === cipherText
+  );
+  return duplicateWallet
+    ? `Duplicate wallet detected: it seems the wallet duplicates the wallet at \n'${duplicateWallet.path}'.`
+    : '';
+};
+export interface WalletByPathWithValidationError extends WalletWithPath {
+  error: string;
+}
+export const validateWalletsForList = (wallets: WalletWithPath[]): string[] =>
```

Comment:
It is not a type predicate (previously known as "type guard"):
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates